### ISSUE TITLE
Ability to disable the desired widgets / Important feature

### ISSUE THREAD

**Issue by 79ho3ein** [elementor/elementor#17466]
- [ ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

Hello

As you know, the number of elementor widgets is increasing day by day. Every website we build does not require at least 60% of the widgets.
Almost all elementor addons have the ability to disable unused widgets. But it is strange that elementor does not address such an important feature.
This factor reduces the number of active widgets, as well as greatly increases the speed of the elemnetor editor and eliminates clutter.
I do not know if elementor has a plan to add this feature or not, but it really needs it.
I hope to see this feature in 2022. I hope.

Thank you

**Comment by SwingWizard** [comment]
This is certainly needed. Many a time, widgets conflict with other plugins frustrate many especially novice users.

**Comment by black-eye** [comment]
Unfortunately Elementor doesn't seem to care about this freature. Almost 3 years ago, I've created the very same feature request (https://github.com/elementor/elementor/issues/7530). After a while it was merged with somehow similar issue which was yet another 2 years older (https://github.com/elementor/elementor/issues/1314). 

In 15. 3. 2019 they started a FB poll for the most wanted feature. This feature has won (hands down) the whole poll (https://www.facebook.com/groups/Elementors/posts/1918292704963173/)

![obrazek](https://user-images.githubusercontent.com/5634870/149195788-44b9fcba-46f8-4dda-9ad5-57ace02aed8c.png)

Still no sign of any progress...

**Comment by paauserrat** [comment]
Looking forward to this!

**Comment by Michaelgimii** [comment]
+1
This is really great. @shilo-ey , @DorShahar  , @avivu Is there any news about this feature? Please let us know

**Comment by alexdrona** [comment]
+1
Isn't it better for developers to poll their users about future features?
Now a new page builder called breaks has been created, whose developers pay attention to their users' opinions. They work fast and conduct a monthly survey to find out what users need. So they quickly add a feature that is in high demand.
With this elementor speed in its development, it will lose its place in less than a year. There are over 10 requests on GitHub that are at least 4 years old. Why should this be so? Has elementor forgotten what it is?
Elemental developers, if you read my message, know that this speed of progress is really, really disappointing. Please think about this. I'm not sure, but we're very worried about that.
If elementor loses its current status and is replaced by another page builder, it will not be easy for us to move to another page builder. We like elementor. And we want it to stay in the first place forever. This goal will not be achieved unless the development of this plugin is accelerated. thanks

**Comment by 79ho3ein** [comment]
hi @alexdrona , This is not so bad and disappointing. elementor releases great updates. Like lementor pro 3.6. But I agree with you. The pace of progress seems a bit slow.

You made a good point. **It would be great if the developers could launch a survey system.** For example, once every 3 months, put 5 features in the poll. Develop the feature that gets the highest votes among users. this is fabulous. Using this feature, high priority features are specified. So users feel better too.

**Comment by sf-steve** [comment]
Whilst it would certainly be great if a UI was added, it is currently possible to use hooks to do so fairly simply:

````
add_action('elementor/widgets/widgets_registered', function(\Elementor\Widgets_Manager $widgets_Manager){

        // /plugins/elementor/includes/managers/widgets.php
        //google maps uses an underscore in the get_name() method, which is what
        //widget manager usues internally, so this array differes from the one in the plugin
        $build_widgets_filename = [
//                'common',
//                'heading',
//                'image',
//                'text-editor',
//                'video',
//                'button',
//                'divider',
//                'spacer',
//                'image-box',
//                'google_maps',
//                'icon',
//                'icon-box',
//                'image-gallery',
//                'image-carousel',
//                'icon-list',
//                'counter',
//                'progress',
//                'testimonial',
//                'tabs',
//                'accordion',
//                'toggle',
//                'social-icons',
//                'alert',
//                'audio',
//                'shortcode',
//                'html',
//                'menu-anchor',
//                'sidebar',
        ];
        foreach ($build_widgets_filename as $widget) {
            $widgets_Manager->unregister_widget_type($widget);
        }
    });
````

**Comment by rami-elementor** [comment]
We have added the snippet to our Developers Docs

https://developers.elementor.com/docs/managers/unregistering-widgets/#remove-unused-widgets

**Comment by 79ho3ein** [comment]
@rami-elementor 
Will this feature be formally added to elementor? If yes, in 2022?

**Comment by black-eye** [comment]
> We have added the snippet to our Developers Docs
> 
> https://developers.elementor.com/docs/managers/unregistering-widgets/#remove-unused-widgets

Really? After 5 years of requesting this feature, you have added a piece of code to your Docs, basically telling us: "If you want the UI for the Widget Manager, code your own"?

The worst part is, that I'm not surprised...

**Comment by 79ho3ein** [comment]
I have always had a question ...
There are many important features that elemntor already has code for, but does not add. Most importantly, there are many important features that elementor add-ons have already added. So their code is fully accessible and ready in advance. I can point out the display conditions feature that 5 elementor add-ons use this feature, but elementor still lacks it. With all the welcome requests that users have for this feature, in this GitHub.
I do not know why elementor development takes so long. There are so many issues that 5 years have passed. Last year I submitted a request to add a border radious feature to the slides widget. But there is no news about it yet and elementor pro in update 3.6 adds the feature of changing the color of navigation points!
I really do not know what's going on in elementor. With this development process, our life will end, but the elementor will not reach the desired conditions.
As a professional elementor user, if I just wanted to write elementor flaws and bugs in the GitHub, I would have to write about them all for 1 week. However, I am willing to do that. Proudly. But I know no attention is paid to them. Personally, I take hours to find the cause of a bug and fully report the steps of reproducing it, but no attention is paid to them.
I have been active in elementar GitHub for a year. Until a month ago, only 1 in 20 issues was reviewed by developers. @ rami-elementor has been reviewing and responding to user suggestions for almost a month, and I're really grateful to he.
If it does not approve a request from us and closes it, it does no problem, at least it pays attention to the users' request and interacts with the users.

I hope one day elementor gets closer to its users. The day that in GitHub asks its users for their opinion, **which feature would you like us to add in the next version?** So the most popular feature will be added in the next version.
I hope that the day will come when elementor will pay special attention to the opinions and ideas of its users in GateHub. Especially to bugs reported by users. Unfortunately, even bug fixes take a long time now...

I hope dear developers do not get upset by what I said. They are really the best, they have made the best website builder in the world. There is no doubt about that. But there is very, very much room for improvement.

Proudly, elementor user

**Comment by rami-elementor** [comment]
> Really? After 5 years of requesting this feature, you have added a piece of code to your Docs, basically telling us: "If you want the UI for the Widget Manager, code your own"?
> 
> The worst part is, that I'm not surprised...

@black-eye please understand, we have 10 million users. Each user sends us his own feature requests vie GitHub, Email, social media, support etc. We need to prioritize those requests and decide which will have the highest impact on our users.

I really want to implement all the great feature requests you send, there are amazing suggestions coming from our users, but we just don't have the man power to do that.

We have to research each request to see how to implements it, will it break existing sites, check the site performance impact, SEO, and more.

We prioritize bugs and regressions. Other feature requests are addon territory. Hope you will understand.

**Comment by black-eye** [comment]
@rami-elementor 
Don't get me wrong, I absolutely understand, you cannot please everyone and you must have some priorities based on your vision. 

What I do not understand is the key which rules these priorities. As I referenced above, 3 years ago (in 2 days excactly) your own representative started a poll on your FB group on the most desired feature, in which this so called widget manager has won (by far). This feature is not some random eye candy redundant stuff - it has won the poll for the most requested feature! Yes, it has not so massive support here on Github, but I believe, it's partly because the requests are splitted across several closed issues and the one left open has a bit unclear name and definition.

Concerning the addon territory. Honestly, what other feature is in the core territory than the one that controls all the features included? Do you advice us to install a separate plugin to control unwanted features of your own plugin? Does it make sense to you?

And one more thing - this feature manager shoudn't be solely about disabling the widgets, it should control all the features in general. I mean stuff like popups, custom code, forms... You don't want them to reduce the bloat, make the backend quicker, or just get rid of too long menu item list? Just turn them off. You are on the right way there with the Experiments settings, but it feels more like an afterthought (and I'm not sure wheather it will stay after the features get to their final stages).

I simply don't get it - you are trying hard to reduce the bloat, but instead of letting us to turn unwanted features off, you are adding more and more dispensable features which are often in the addon territory (Text Path, come on...). Each of such feature adds CSS, prolong the loading time and raises the memory demand (causing the dreaded infinite loading issue). I know it, I've been there on some less equipped hostings, where I had to use unregisterring the widgets one by one to make the editor work. You are simply against yourself not implementong this functionality. I don't get it.

Sorry for such a long post, but I hope it will be received as constructive as I meant it. To be honest my previous rant was caused by sheer frustration from Elementor silence on simiral topics. I really appreciate you replied to it. To be fair, you are the one that replies always on the topic and I feel your you are sincerely trying to be helpful especially on more technical issues. My objections were not against you personally. :-)

**Comment by rami-elementor** [comment]
Thank you for the reply, I'll make sure the decision makers will read it.

As for the poll results, we have seen it. Instead of removing widgets, which can have side-effects on pages using those widgets, we decided to added the "Favorite" feature where you can set widgets as favorites.

Also keep in mind that removing unused widgets won't "reduce the bloat" and won't make your site run faster, it will just make the [Widgets Panel](https://developers.elementor.com/docs/editor/widgets-panel/) inside [Elementor Editor](https://developers.elementor.com/docs/editor/) customizable.

**Comment by 79ho3ein** [comment]
@rami-elementor , Removing unnecessary widgets dramatically increases the speed of the elementor editor. I've tested this before on other plugins that have this feature.

**Comment by rami-elementor** [comment]
@79ho3ein We are working on a better solution. We want to rewrite the entire Editor with performance in mind. It should solve many of the issue users currently have. The current Editor was developed over 5 years ago and it has many drawbacks. But it's a long term objective.

**Comment by pauserratgutierrez** [comment]
> Thank you for the reply, I'll make sure the decision makers will read it.
> 
> As for the poll results, we have seen it. Instead of removing widgets, which can have side-effects on pages using those widgets, we decided to added the "Favorite" feature where you can set widgets as favorites.
> 
> Also keep in mind that removing unused widgets won't "reduce the bloat" and won't make your site run faster, it will just make the [Widgets Panel](https://developers.elementor.com/docs/editor/widgets-panel/) inside [Elementor Editor](https://developers.elementor.com/docs/editor/) customizable.

I understand but disabling widgets should not be a problem if done right. You can implement diferent methods to detect if the page was using a disabled widget.

- When trying to edit it an alert message can pop up if it detects that a disabled widget is in use.
- When disabling something a check can be done to see if the widget is in use and alert the user.

**Comment by pauserratgutierrez** [comment]
> @79ho3ein We are working on a better solution. We want to rewrite the entire Editor with performance in mind. It should solve many of the issue users currently have. The current Editor was developed over 5 years ago and it has many drawbacks. But it's a long term objective. 

Oh cool, that's a good thing!

**Comment by rami-elementor** [comment]
> I understand but disabling widgets should not be a problem if done right. You can implement diferent methods to detect if the page was using a disabled widget.

Again, those things require planning as we need to identify all the possible issues and develop backwards compatible solutions.

**Comment by 79ho3ein** [comment]
@rami-elementor , Thank you very much for your attention to user comments and ideas. 
Definitely rewriting the editor will fix all the problems. Thanks again

**Comment by rami-elementor** [comment]
@79ho3ein I will close this issue, it's a duplicate of an older issue.

**Comment by rami-elementor** [comment]
Duplicate of #9647

**Comment by nicholaszein** [comment]
Hey guys!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with Elementor v3.17.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
